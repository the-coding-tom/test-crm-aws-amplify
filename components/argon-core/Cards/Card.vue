<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
