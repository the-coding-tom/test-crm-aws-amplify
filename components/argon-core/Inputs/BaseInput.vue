<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
