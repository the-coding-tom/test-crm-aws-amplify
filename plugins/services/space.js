export default function(ctx, inject) {
  const Space = {
    updateOneSpace: (spaceId, payload) => {
      return ctx.$axios.$patch(`/spaces/${spaceId}`, payload)
    }
  }
  ;(ctx.$space = Space), inject('space', Space)
}
