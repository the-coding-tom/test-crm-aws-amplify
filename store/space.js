export const state = () => ({
  currentSpace: null
})

export const mutations = {
  setSpace(state, spaceDetails) {
    state.currentSpace = spaceDetails
  }
}
