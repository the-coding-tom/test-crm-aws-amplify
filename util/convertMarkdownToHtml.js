export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
