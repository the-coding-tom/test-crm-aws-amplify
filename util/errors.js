export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
