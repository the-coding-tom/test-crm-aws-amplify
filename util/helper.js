const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
