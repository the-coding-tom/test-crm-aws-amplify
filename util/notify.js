const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
