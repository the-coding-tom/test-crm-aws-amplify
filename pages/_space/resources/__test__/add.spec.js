import AddPage from '@/pages/_space/resources/add.vue'
import Card from '~/components/argon-core/Cards/Card.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BootstrapVue from 'bootstrap-vue'
import BaseHeader from '~/components/argon-core/BaseHeader'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'
import moment from 'moment'

import space from '@/pages/_space/__test__/resources/current-space.json'
import newCategory from './resources/new-resource.json'
import allZoomRooms from './resources/zoom-rooms.json'
import allCategories from './resources/all-categories.json'
import allAccessPoints from './resources/access-points.json'

describe('Page: /:space/resources/ | Resources Add Page', () => {
  let NuxtStore, store, wrapper, router, $resource, $accesspoint, $zoomrooms

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $zoomrooms = {
      getRooms: jest.fn().mockResolvedValue(cloneDeep(allZoomRooms))
    }

    $resource = {
      getAllCategories: jest.fn().mockResolvedValue(cloneDeep(allCategories)),
      createRoom: jest.fn().mockResolvedValue()
    }

    $accesspoint = {
      list: jest.fn().mockResolvedValue(cloneDeep(allAccessPoints))
    }
  })

  beforeEach(async () => {
    router = new VueRouter({
      routes: []
    })

    store = await NuxtStore.createStore()
    store.commit('space/setSpace', space)
    store.$resource = $resource
    store.$router = router
    store.$moment = moment
    store._vm.$bvToast = {
      toast: jest.fn()
    }

    const wrapperAsyncData = await AddPage.asyncData({
      store,
      $zoomrooms,
      $accesspoint
    })

    wrapper = createWrapper(
      AddPage,
      store,
      router,
      {
        loading: false,
        ...wrapperAsyncData
      },
      {
        stubs: {
          NuxtLink: true,
          transition: false
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(BaseHeader.name, BaseHeader)
        vueInstance.component(Card.name, Card)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.prototype.$accesspoint = $accesspoint
        vueInstance.prototype.$zoomrooms = $zoomrooms
        vueInstance.prototype.$moment = moment
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('adds a resource', async () => {
    // fill input fields
    wrapper.find('#name').setValue(newCategory.name)
    wrapper.find('#pricePerHour').setValue(newCategory.pricePerHour)
    wrapper.find('#capacity').setValue(newCategory.capacity)
    wrapper.find('#description').setValue(newCategory.description)
    wrapper.find('#minBookingDuration').setValue(newCategory.minBookingDuration)
    wrapper.find('#maxBookingDuration').setValue(newCategory.maxBookingDuration)
    wrapper
      .find('#availableBookingTime')
      .setValue(newCategory.availableBookingTime)
    wrapper.find('#canBook').setChecked(true)

    //  select category
    const categoryOptions = wrapper.vm.$el.querySelectorAll(
      '.el-select-dropdown__item'
    )
    const evt = document.createEvent('MouseEvents')
    evt.initEvent('mouseenter')
    categoryOptions[1].dispatchEvent(evt)
    categoryOptions[1].click()
    await flushPromises()

    // select zoom id
    const zoomOptions = wrapper.findAll('#accessPointId option')
    zoomOptions.at(1).setSelected()
    await flushPromises()

    // select accesspoint
    const accessPointOptions = wrapper.findAll('#zoomRoomId option')
    accessPointOptions.at(1).setSelected()
    await flushPromises()

    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    expect($resource.createRoom).toHaveBeenCalled()
    expect($resource.createRoom.mock.calls[0][0].name).toBeDefined()
    expect($resource.createRoom.mock.calls[0][0].description).toBeDefined()
    expect($resource.createRoom.mock.calls[0][0].capacity).toBeDefined()
    expect(
      $resource.createRoom.mock.calls[0][0].min_booking_duration
    ).toBeDefined()
    expect(
      $resource.createRoom.mock.calls[0][0].max_booking_duration
    ).toBeDefined()
    expect($resource.createRoom.mock.calls[0][0].price_per_hour).toBeDefined()
    expect($resource.createRoom.mock.calls[0][0].can_book).toBeDefined()
    expect($resource.createRoom.mock.calls[0][0].room_category_id).toBeDefined()
    expect(
      $resource.createRoom.mock.calls[0][0].room_availability
    ).toBeDefined()
    expect($resource.createRoom.mock.calls[0][0].zoom_room_id).toBeDefined()
    expect($resource.createRoom.mock.calls[0][0].access_point_id).toBeDefined()
  })
})
