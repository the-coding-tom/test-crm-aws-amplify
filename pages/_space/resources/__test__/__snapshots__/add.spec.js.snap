// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/resources/ | Resources Add Page renders properly 1`] = `
<div>
  <form class="">
    <div class="header pb-6">
      <div class="container-fluid">
        <div class="header-body">
          <div class="d-flex justify-content-between py-4">
            <h1 class="page-title"><span>Add New Resource</span> <b style="display: none;">•</b> </h1> <button type="submit" class="btn btn-primary">
              Save Resource
            </button>
          </div>
        </div>
      </div>
    </div>
    <div class="container-fluid mt--6">
      <div class="card-deck flex-column flex-xl-row">
        <div class="card">
          <!---->
          <div class="card-body">
            <h3 class="mr-b-20">Resource Details</h3>
            <div class="row">
              <div class="col-md-6">
                <div class="row pd-r-20">
                  <div class="form-group col-md-6"><label class="form-control-label">
                      Name
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="text" valid="true" required="required" id="name" placeholder="Name of Resource" class="form-control">
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </div>
                  <div class="form-group col-md-6"><label>Category</label>
                    <div class="el-select" required="required">
                      <!---->
                      <div class="el-input el-input--suffix">
                        <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="请选择" name="category" id="category" class="el-input__inner">
                        <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----><!----><!----><!----><!----></span>
                        <!----></span>
                        <!---->
                        <!---->
                      </div>
                      <div class="el-select-dropdown el-popper" style="display: none;">
                        <div class="el-scrollbar" style="">
                          <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                            <ul class="el-scrollbar__view el-select-dropdown__list">
                              <!---->
                              <li class="el-select-dropdown__item"><span>conference</span></li>
                              <li class="el-select-dropdown__item"><span>meeting</span></li>
                            </ul>
                          </div>
                          <div class="el-scrollbar__bar is-horizontal">
                            <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                          </div>
                          <div class="el-scrollbar__bar is-vertical">
                            <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                          </div>
                        </div>
                        <!---->
                      </div>
                    </div>
                  </div>
                  <div class="form-group col-md-6"><label class="form-control-label">
                      Price per hour
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="number" valid="true" required="required" id="pricePerHour" step="0.01" placeholder="0.00" class="form-control">
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </div>
                  <div class="col-md-6">
                    <div class="form-group"><label class="form-control-label">
                        Number of Units
                      </label>
                      <div class="input-group has-label">
                        <!----> <input type="number" valid="true" required="required" id="capacity" placeholder="0" class="form-control">
                        <!---->
                        <!---->
                        <!---->
                      </div>
                    </div> <small class="form-text text-muted">How many people can book this at the same time</small>
                  </div>
                  <div class="form-group col-md-12"><label>Description</label> <textarea id="description" placeholder="Add details about this resource" rows="4" max-rows="6" required="required" class="form-control"></textarea></div>
                  <div class="form-group col-md-6"><label class="form-control-label">
                      Minimum Booking Duration
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="number" valid="true" required="required" id="minBookingDuration" min="0" placeholder="0" class="form-control">
                      <div class="input-group-append"><span class="input-group-text"><div>min</div></span></div>
                      <!---->
                      <!---->
                    </div>
                  </div>
                  <div class="form-group col-md-6"><label class="form-control-label">
                      Maximum Booking Duration
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="number" valid="true" required="required" id="maxBookingDuration" placeholder="0" class="form-control">
                      <div class="input-group-append"><span class="input-group-text"><div>max</div></span></div>
                      <!---->
                      <!---->
                    </div>
                  </div>
                </div>
              </div>
              <div class="col-md-6">
                <div class="row pd-l-20">
                  <div class="form-group col-md-12"><label class="form-control-label">
                      Available Booking Time
                    </label>
                    <div class="input-group has-label">
                      <!----> <input type="text" valid="true" required="required" id="availableBookingTime" placeholder="0" class="form-control">
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </div> <small class="col-md-12 form-text text-muted">You can enter multiple time frames separated by comma, e.g. e.g. mo-fr 9-17, sa 10-3.</small>
                  <div class="col-md-12 form-group"><label>Amenities</label>
                    <div class="tags-input__wrapper" id="amenities" type="Text" required="required"> <input type="text" placeholder="Add amenities" size="mini" class="form-control"></div>
                  </div>
                  <fieldset class="form-group col-md-12" id="__BVID__25">
                    <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__25__BV_label_">Zoom Rooms</legend>
                    <div tabindex="-1" role="group" class="bv-no-focus-ring"><select id="zoomRoomId" class="custom-select">
                        <option value="sqxuaM7URRK3CuNQWH2lra">001 ORD</option>
                        <option value="Gasam9e9iTZSTEdZXL0PnJg">021 HJORD</option>
                        <option value="">Choose a zoom room</option>
                      </select>
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </fieldset>
                  <fieldset class="form-group col-md-12" id="__BVID__27">
                    <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__27__BV_label_">Brivo Access Point</legend>
                    <div tabindex="-1" role="group" class="bv-no-focus-ring"><select id="accessPointId" class="custom-select">
                        <option value="24633465">001 Geirangerfjord</option>
                        <option value="24633585">002 Hardangerfjord</option>
                        <option value="">Choose a zoom room</option>
                      </select>
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </fieldset>
                  <div class="form-group col-md-12"><label>Resource Settings</label>
                    <div class="custom-control custom-checkbox"><input id="canBook" type="checkbox" autocomplete="off" class="custom-control-input" value="true"><label for="canBook" class="custom-control-label">Room is bookable</label></div>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div id="preview" class="mr-b-20">
                <!---->
              </div>
              <div class="bg-lgray pd-10 pd-l-20 upload-btn-wrapper pd-0"><button class="btn"><i class="fa fa-paperclip mr-r-10"></i> Upload File (&lt;500KB &amp; size 1125x582)</button> <input type="file" accept="image/*"></div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </form>
</div>
`;
